[FIO internal] Revert "core: introduce CFG_CORE_HUK_SUBKEY_COMPAT_USE…

ldts opened the pull request:
…_OTP_DIE_ID"

Modifying the FEK breaks secure storage on OTA

https://github.com/OP-TEE/optee_os/issues/7327

This reverts commit 33b38f8c843ad4bd450639c82ac77babeff581e6.

Signed-off-by: jorge@foundries.io

<!--
    If you are new to submitting pull requests to OP-TEE, then please have a
    look at the list below and tick them off before submitting the pull request.

    1. Read our contribution guidelines:
         https://optee.readthedocs.io/en/latest/general/contribute.html

    2. Read the contribution section in Notice.md and pay extra attention to the
       "Developer Certificate of Origin" in the contribution guidelines.

    3. You should run checkpatch preferably before submitting the pull request.

    4. When everything has been reviewed, you will need to squash, rebase and
       add tags like `Reviewed-by`, `Acked-by`, `Tested-by` etc. More details
       about this can also be found on the link provided above.

    NOTE: This comment will not be shown in the pull request, so no harm keeping
    it, but feel free to remove it if you like.
-->

ricardosalveti commented on core/arch/arm/kernel/thread.c:
```diff
@@ -97,12 +97,18 @@ linkage uint32_t name[num_stacks] \
+#ifdef CFG_NXP_SE05X_SVC
+DECLARE_STACK(stack_tmp, CFG_TEE_CORE_NB_CORE, 5 * STACK_TMP_SIZE, static);
+#ifndef CFG_WITH_PAGER
+DECLARE_STACK(stack_thread, CFG_NUM_THREADS, 2 * STACK_THREAD_SIZE, static);
```
How did you get to these values?

ldts commented:
just trial and error until it came to a number that it booted (didnt spend
the time doing a stack analysis at this point)

On Wed, Jul 8, 2020 at 12:44 AM Ricardo Salveti <notifications@github.com>
wrote:

> *@ricardosalveti* commented on this pull request.
> ------------------------------
>
> In core/arch/arm/kernel/thread.c
> <https://github.com/foundriesio/optee_os/pull/10#discussion_r451184949>:
>
> >  DECLARE_STACK(stack_abt, CFG_TEE_CORE_NB_CORE, STACK_ABT_SIZE, static);
> +#ifdef CFG_NXP_SE05X_SVC
> +DECLARE_STACK(stack_tmp, CFG_TEE_CORE_NB_CORE, 5 * STACK_TMP_SIZE, static);
> +#ifndef CFG_WITH_PAGER
> +DECLARE_STACK(stack_thread, CFG_NUM_THREADS, 2 * STACK_THREAD_SIZE, static);
>
> How did you get to these values?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/foundriesio/optee_os/pull/10#pullrequestreview-444299933>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACPYXVP5MROXHL6N6GOPST3R2OQMDANCNFSM4OLMJQZA>
> .
>

ldts committed:
[FIO internal] Revert "core: introduce CFG_CORE_HUK_SUBKEY_COMPAT_USE_OTP_DIE_ID"

Modifying the FEK breaks secure storage on OTA

https://github.com/OP-TEE/optee_os/issues/7327

This reverts commit 33b38f8c843ad4bd450639c82ac77babeff581e6.

Signed-off-by: jorge@foundries.io

ricardosalveti commented:
I prefer us changing the config instead in lmp, as then we don't have to keep another patch here, let me propose something to meta-lmp.

ricardosalveti commented:
https://github.com/foundriesio/meta-lmp/pull/1620
